File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Holy_Smite.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import com.sk89q.worldguard.bukkit.WGBukkit;
import com.sk89q.worldguard.protection.flags.DefaultFlag;
import com.sk89q.worldguard.protection.flags.StateFlag;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

import java.util.Iterator;

public class Holy_Smite implements Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    public void onDamage(final EntityDamageByEntityEvent entityDamageByEntityEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "holy_smite"));
        if (entityDamageByEntityEvent.getEntity() instanceof Player && entityDamageByEntityEvent.getDamager() instanceof Player) {
            try {
                final Player player = (Player) entityDamageByEntityEvent.getEntity();
                final Player player2 = (Player) entityDamageByEntityEvent.getDamager();
                if (entityDamageByEntityEvent.isCancelled()) {
                    return;
                }
                if (SettingsManager.enchant.getBoolean("allow-worldguard") && WGBukkit.getRegionManager(player.getWorld()).getApplicableRegions(player.getLocation()).queryState(null, new StateFlag[]{DefaultFlag.PVP}) == StateFlag.State.DENY) {
                    return;
                }
                final int n = (int) (Math.random() * 100.0);
                if (player2.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && n < SettingsManager.enchant.getInt("holy_smite.level_I.chance")) {
                    player.getWorld().strikeLightningEffect(player.getLocation());
                    final Iterator iterator = player.getActivePotionEffects().iterator();
                    while (iterator.hasNext()) {
                        player.removePotionEffect(((PotionEffect) iterator.next()).getType());
                    }
                }
                if (player2.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " II") && n < SettingsManager.enchant.getInt("holy_smite.level_II.chance")) {
                    player.getWorld().strikeLightningEffect(player.getLocation());
                    final Iterator iterator2 = player.getActivePotionEffects().iterator();
                    while (iterator2.hasNext()) {
                        player.removePotionEffect(((PotionEffect) iterator2.next()).getType());
                    }
                }
                if (player2.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " III") && n < SettingsManager.enchant.getInt("holy_smite.level_III.chance")) {
                    player.getWorld().strikeLightningEffect(player.getLocation());
                    final Iterator iterator3 = player.getActivePotionEffects().iterator();
                    while (iterator3.hasNext()) {
                        player.removePotionEffect(((PotionEffect) iterator3.next()).getType());
                    }
                }
                if (player2.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " IV") && n < SettingsManager.enchant.getInt("holy_smite.level_IV.chance")) {
                    player.getWorld().strikeLightningEffect(player.getLocation());
                    final Iterator iterator4 = player.getActivePotionEffects().iterator();
                    while (iterator4.hasNext()) {
                        player.removePotionEffect(((PotionEffect) iterator4.next()).getType());
                    }
                }
                if (player2.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " V") && n < SettingsManager.enchant.getInt("holy_smite.level_V.chance")) {
                    player.getWorld().strikeLightningEffect(player.getLocation());
                    final Iterator iterator5 = player.getActivePotionEffects().iterator();
                    while (iterator5.hasNext()) {
                        player.removePotionEffect(((PotionEffect) iterator5.next()).getType());
                    }
                }
            } catch (Exception ex) {
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantments/Holy_Smite.kt
package org.pixeltime.enchantmentsenhance.event.enchantment

import com.sk89q.worldguard.bukkit.WGBukkit
import com.sk89q.worldguard.protection.flags.DefaultFlag
import com.sk89q.worldguard.protection.flags.StateFlag
import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.potion.PotionEffect
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Holy_Smite : Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    fun onDamage(entityDamageByEntityEvent: EntityDamageByEntityEvent) {
        val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "holy_smite"))
        if (entityDamageByEntityEvent.entity is Player && entityDamageByEntityEvent.damager is Player) {
            try {
                val player = entityDamageByEntityEvent.entity as Player
                val player2 = entityDamageByEntityEvent.damager as Player
                if (entityDamageByEntityEvent.isCancelled) {
                    return
                }
                if (SettingsManager.enchant.getBoolean("allow-worldguard") && WGBukkit.getRegionManager(player.world).getApplicableRegions(player.location).queryState(null, *arrayOf(DefaultFlag.PVP)) == StateFlag.State.DENY) {
                    return
                }
                val n = (Math.random() * 100.0).toInt()
                if (player2.itemInHand.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " I") && n < SettingsManager.enchant.getInt("holy_smite.level_I.chance")) {
                    player.world.strikeLightningEffect(player.location)
                    val iterator = player.activePotionEffects.iterator()
                    while (iterator.hasNext()) {
                        player.removePotionEffect((iterator.next() as PotionEffect).type)
                    }
                }
                if (player2.itemInHand.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " II") && n < SettingsManager.enchant.getInt("holy_smite.level_II.chance")) {
                    player.world.strikeLightningEffect(player.location)
                    val iterator2 = player.activePotionEffects.iterator()
                    while (iterator2.hasNext()) {
                        player.removePotionEffect((iterator2.next() as PotionEffect).type)
                    }
                }
                if (player2.itemInHand.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " III") && n < SettingsManager.enchant.getInt("holy_smite.level_III.chance")) {
                    player.world.strikeLightningEffect(player.location)
                    val iterator3 = player.activePotionEffects.iterator()
                    while (iterator3.hasNext()) {
                        player.removePotionEffect((iterator3.next() as PotionEffect).type)
                    }
                }
                if (player2.itemInHand.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " IV") && n < SettingsManager.enchant.getInt("holy_smite.level_IV.chance")) {
                    player.world.strikeLightningEffect(player.location)
                    val iterator4 = player.activePotionEffects.iterator()
                    while (iterator4.hasNext()) {
                        player.removePotionEffect((iterator4.next() as PotionEffect).type)
                    }
                }
                if (player2.itemInHand.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " V") && n < SettingsManager.enchant.getInt("holy_smite.level_V.chance")) {
                    player.world.strikeLightningEffect(player.location)
                    val iterator5 = player.activePotionEffects.iterator()
                    while (iterator5.hasNext()) {
                        player.removePotionEffect((iterator5.next() as PotionEffect).type)
                    }
                }
            } catch (ex: Exception) {
            }

        }
    }
}
